pkg-manager/resolve-dependencies/src/resolveDependencies.ts

```diff
@@ -537,6 +537,9 @@ export async function resolveDependencies (
       newPreferredVersions[resolvedPackage.name] = {}
     }
```

Comment:
instead of the change below, this can be replaced with 

```js
    if (!Object.prototype.hasOwnProperty.call(newPreferredVersions, resolvedPackage.name)) {
      newPreferredVersions[resolvedPackage.name] = preferredVersions[resolvedPackage.name]
        ? Object.create(preferredVersions[resolvedPackage.name])
        : {}
    }
```

which is noticeably faster - 2m 50s vs 3m 16.8s for me

Reply:
But in npm-resolver we are then running `Object.entries` and `Object.keys` on `newPreferredVersions[resolvedPackage.name]`, which will not list the properties if they were added with `Object.create`

Reply:
indeed that can be a problem, let's stick to safer option creating a copy